fiber/chain_interactions/commitments.py

```diff
@@ -0,0 +1,224 @@
+from enum import Enum
+from typing import TypeAlias
+
+from pydantic import BaseModel
+from scalecodec import ScaleType
+from substrateinterface import SubstrateInterface, Keypair
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from fiber.chain_interactions.chain_utils import format_error_message
+
+
+class DataFieldType(Enum):
+    RAW = "Raw"
+    BLAKE_TWO_256 = "BlakeTwo256"
+    SHA_256 = "Sha256"
+    KECCAK_256 = "Keccak256"
+    SHA_THREE_256 = "ShaThree256"
+
+
+CommitmentDataField: TypeAlias = tuple[DataFieldType, bytes] | None
+
+
+class CommitmentQuery(BaseModel):
+    fields: list[CommitmentDataField]
+    block: int
+    deposit: int
+
+
+class RawCommitment(BaseModel):
+    data: bytes
+    block: int
+    deposit: int
+
+
+def _serialize_field(field: CommitmentDataField):
+    if not field:
+        return {str(None): b''}
```

Comment:
str(None) 0.o 

should probably be a constant in fiber.constants.py or similar, of 'None' if we want to return that [im guessing this is straight from BT?]

Reply:
Yep this is the field data type `None`, moved it to constants